core/state/statedb.go

```diff
@@ -294,9 +295,12 @@ func (s *StateDB) GetCodeSize(addr common.Address) int {
 	if stateObject.code != nil {
 		return len(stateObject.code)
 	}
+	if bytes.Equal(stateObject.CodeHash(), emptyCode[:]) {
+		return 0
+	}
 	size, err := s.db.ContractCodeSize(stateObject.addrHash, common.BytesToHash(stateObject.CodeHash()))
 	if err != nil {
-		s.setError(err)
+		s.setError(fmt.Errorf("GetCodeSize (%x) error: %v", addr[:], err))
 	}
 	return size
 }
```

Comment:
If we rewrote GetCodeSize to be similar to GetCode, it we could avoid having to handle the same special case (.code being unloaded) twice and instead rely on the same mechanism Getcode does (obj.Code):

```go
func (s *StateDB) GetCodeSize(addr common.Address) []byte {
    stateObject := s.getStateObject(addr)
    if stateObject != nil {
        return len(stateObject.Code(s.db))
    }
    return 0
}
```

Reply:
Actually, why do we even have this method? Can't we just `len(statedb.Code(addr))`? We need to pull the code up anyway.